google/cloud/bigtable/tests/data_integration_test.cc

```diff
@@ -122,6 +123,28 @@ TEST_P(DataIntegrationTest, TableBulkApply) {
   CheckEqualUnordered(expected, actual);
 }
 
+TEST_P(DataIntegrationTest, TableBulkApplyThrottling) {
+  if (GetParam() == "with-data-client") GTEST_SKIP();
+
+  // Make a custom table with throttling enabled.
+  auto table = Table(
+      MakeDataConnection(Options{}.set<experimental::BulkApplyThrottlingOption>(
+          absl::monostate{})),
+      TableResource(TableTestEnvironment::project_id(),
+                    TableTestEnvironment::instance_id(),
+                    TableTestEnvironment::table_id()));
+
+  // This test will take at least 100 queries / (20 QPS) = 5s.
```

Comment:
Perhaps we should be able to inject a "sleeper" via the options.

Reply:
Ack. We could also pass `MockConnection`s to `Make*Connection(Options const& options)` via the options for better testing of those functions.

Although, at some point we are going to need an integration test for the real implementation.

Reply:
OK, but if tests start sleeping for 5s here and there, we're only hurting ourselves.

Reply:
Reduced from 100 -> 10 requests.